engine/schema/src/com/cloud/upgrade/DatabaseUpgradeChecker.java

```diff
@@ -408,6 +408,9 @@ public DatabaseUpgradeChecker() {
         _upgradeMap.put(CloudStackVersion.parse("4.9.3.0"),
             new DbUpgrade[] {new Upgrade4930to41000(), new Upgrade41000to41100(), new Upgrade41100to41110()});
 
+        _upgradeMap.put(CloudStackVersion.parse("4.9.3.1"),
+                new DbUpgrade[] {new Upgrade4930to41000(), new Upgrade41000to41100(), new Upgrade41100to41110()});
```

Comment:
Hi @rafaelweingartner, Thanks for adding this upgrade path. Here we are considering from 4930 to 41000 instead of 4931 to 41000. Can we write a test for this path in DatabaseUpgradeCheckerTest.

Reply:
Yes, plus a schema file if there is any schema change. And adding a unit test in the DatabaseUpgradeCheckerTest.

Reply:
This is fine, the security release the 4th dot number (a.b.c.x, the `x`) is considered to NOT usually have any db path/upgrade change, which is why we introduced it in the past. The change is acceptable, not requiring tests if manually @rafaelweingartner or someone else can confirm the change.

Reply:
It might not be needed, but it would surely help reading the code and understanding upgrade path easier, specially when we move forward with #2584 , then we wouldn't see duplicate (and redundant) upgrade path. I would suggest to rename java and sql `4930to49100` files to `4930to4931` and create new/empty `4931to49100`.

Reply:
@khos2ow 49100 == 41000?

Reply:
@DaanHoogland YES! Good catch!

Reply:
Changes are fine and acceptable as it is.

Reply:
@rhtyd why confused emoji though?

Reply:
For the purpose of the fix and following the 4.11/legacy db-upgrade model, the changes seem alright to me.

Reply:
Fair enough :+1: